samples/go/ipfs-chat/main.go

```diff
@@ -55,10 +58,8 @@ func main() {
 	daemonCmd := kingpin.Command("daemon", "runs a daemon that simulates filecoin, eagerly storing all chunks for a chat")
 	daemonTopic := daemonCmd.Flag("topic", "IPFS pubsub topic to publish and subscribe to").Default("ipfs-chat").String()
 	daemonInterval := daemonCmd.Flag("interval", "amount of time to wait before publishing state to network").Default("5s").Duration()
-	netNodeIdx := daemonCmd.Flag("net-node-idx", "a single digit to be used as last digit in all port values: api, gateway and swarm (must be 0-9 inclusive)").Default("-1").Int()
-	localNodeIdx := daemonCmd.Flag("local-node-idx", "a single digit to be used as last digit in all port values: api, gateway and swarm (must be 0-9 inclusive)").Default("-1").Int()
-	daemonNetworkDS := daemonCmd.Arg("network-dataset", "the dataset spec to use to read and write data to the IPFS network").Required().String()
-	daemonLocalDS := daemonCmd.Arg("local-dataset", "the dataset spec to use to read and write data locally").Required().String()
+	daemonNodeIdx := daemonCmd.Flag("node-idx", "a single digit to be used as last digit in all port values: api, gateway and swarm (must be 0-9 inclusive)").Default("-1").Int()
```

Comment:
Why not just default it to zero so that you can use its default value, rather than testing whether it has the default value?

Reply:
I would do that if the default configuration had all ports ending in 0. The default config has a mixture of 0's and 1's and so I wanted to leave the defaults unchanged unless a portIdx was specified.